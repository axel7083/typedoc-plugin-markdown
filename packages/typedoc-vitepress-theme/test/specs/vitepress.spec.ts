import * as fs from 'fs';
import * as path from 'path';
describe(`VitePress`, () => {
  test(`should output docs with vitepress theme`, async () => {
    const contents = fs
      .readFileSync(path.join(__dirname, '../out/default/index.md'))
      .toString();
    expect(contents).toMatchSnapshot();
  });

  test(`should output members that require anchor slugification`, async () => {
    const contents = fs
      .readFileSync(
        path.join(
          __dirname,
          '../out/default/module-1/interfaces/InterfaceA.md',
        ),
      )
      .toString();
    expect(contents).toMatchSnapshot();
  });

  test(`should generate typedoc sidebar`, async () => {
    const contents = fs
      .readFileSync(path.join(__dirname, '../out/default/typedoc-sidebar.json'))
      .toString();
    expect(contents).toMatchSnapshot();
  });

  test(`should generate typedoc sidebar with options`, async () => {
    const contents = fs
      .readFileSync(
        path.join(__dirname, '../out/sidebar-options/typedoc-sidebar.json'),
      )
      .toString();
    expect(contents).toMatchSnapshot();
  });
});
