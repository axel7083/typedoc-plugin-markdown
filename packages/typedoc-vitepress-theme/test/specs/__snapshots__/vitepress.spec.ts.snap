// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VitePress should generate typedoc sidebar 1`] = `"[{"text":"module-1","link":"/test/out/default/module-1/index.md","collapsed":true,"items":[{"text":"Classes","collapsed":true,"items":[{"text":"ClassA","link":"/test/out/default/module-1/classes/ClassA.md"},{"text":"ClassB","link":"/test/out/default/module-1/classes/ClassB.md"}]},{"text":"Interfaces","collapsed":true,"items":[{"text":"InterfaceA","link":"/test/out/default/module-1/interfaces/InterfaceA.md"},{"text":"InterfaceB","link":"/test/out/default/module-1/interfaces/InterfaceB.md"}]}]},{"text":"module-2","link":"/test/out/default/module-2/index.md","collapsed":true,"items":[{"text":"Classes","collapsed":true,"items":[{"text":"ClassA","link":"/test/out/default/module-2/classes/ClassA.md"},{"text":"ClassB","link":"/test/out/default/module-2/classes/ClassB.md"}]},{"text":"Interfaces","collapsed":true,"items":[{"text":"InterfaceA","link":"/test/out/default/module-2/interfaces/InterfaceA.md"},{"text":"InterfaceB","link":"/test/out/default/module-2/interfaces/InterfaceB.md"}]}]}]"`;

exports[`VitePress should generate typedoc sidebar with options 1`] = `
"[
  {
    "text": "module-1",
    "link": "/sidebar-options/module-1/index.md",
    "collapsed": false,
    "items": [
      {
        "text": "Classes",
        "collapsed": false,
        "items": [
          {
            "text": "ClassA",
            "link": "/sidebar-options/module-1/classes/ClassA.md"
          },
          {
            "text": "ClassB",
            "link": "/sidebar-options/module-1/classes/ClassB.md"
          }
        ]
      },
      {
        "text": "Interfaces",
        "collapsed": false,
        "items": [
          {
            "text": "InterfaceA",
            "link": "/sidebar-options/module-1/interfaces/InterfaceA.md"
          },
          {
            "text": "InterfaceB",
            "link": "/sidebar-options/module-1/interfaces/InterfaceB.md"
          }
        ]
      }
    ]
  },
  {
    "text": "module-2",
    "link": "/sidebar-options/module-2/index.md",
    "collapsed": false,
    "items": [
      {
        "text": "Classes",
        "collapsed": false,
        "items": [
          {
            "text": "ClassA",
            "link": "/sidebar-options/module-2/classes/ClassA.md"
          },
          {
            "text": "ClassB",
            "link": "/sidebar-options/module-2/classes/ClassB.md"
          }
        ]
      },
      {
        "text": "Interfaces",
        "collapsed": false,
        "items": [
          {
            "text": "InterfaceA",
            "link": "/sidebar-options/module-2/interfaces/InterfaceA.md"
          },
          {
            "text": "InterfaceB",
            "link": "/sidebar-options/module-2/interfaces/InterfaceB.md"
          }
        ]
      }
    ]
  }
]"
`;

exports[`VitePress should output docs with vitepress theme 1`] = `
"# typedoc-vitepress-theme

## Modules

- [module-1](module-1/index.md)
- [module-2](module-2/index.md)
"
`;

exports[`VitePress should output members that require anchor slugification 1`] = `
"# Interface: InterfaceA

See [InterfaceB._prop_with_underscore_](InterfaceB.md#prop-with-underscore)
"
`;
